tests/unit/test_routes.py

```diff
@@ -9,54 +9,231 @@
+def test_get_resources(module_client, module_db):
+    client = module_client
 
-def test_create_resource(session_app, function_session, session_db):
-    client = session_app.test_client()
+    response = client.get('api/v1/resources')
+
+    # Status should be OK
+    assert (response.status_code == 200)
+
+    resources = response.json
+
+    # Default page size shouold be specified in PaginatorConfig
+    assert (len(resources['data']) == PaginatorConfig.per_page)
+
+    for resource in resources['data']:
+        assert (isinstance(resource.get('name'), str))
+        assert (resource.get('name') != "")
+        assert (isinstance(resource.get('url'), str))
+        assert (resource.get('url') != "")
+        assert (isinstance(resource.get('category'), str))
+        assert (resource.get('category') != "")
+        assert (isinstance(resource.get('languages'), list))
+
+
+def test_get_single_resource(module_client, module_db):
+    client = module_client
+
+    response = client.get('api/v1/resources/5')
+
+    # Status should be OK
+    assert (response.status_code == 200)
+
+    resource = response.json['data']
+    assert (isinstance(resource.get('name'), str))
+    assert (resource.get('name') != "")
+    assert (isinstance(resource.get('url'), str))
+    assert (resource.get('url') != "")
+    assert (isinstance(resource.get('category'), str))
+    assert (resource.get('category') != "")
+    assert (isinstance(resource.get('languages'), list))
+
+    assert (resource.get('id') == 5)
+
+
+def test_paginator(module_client, module_db):
+    client = module_client
+
+    # Test page size
+    response = client.get('api/v1/resources?page_size=1')
+    assert (len(response.json['data']) == 1)
+    response = client.get('api/v1/resources?page_size=5')
+    assert (len(response.json['data']) == 5)
+    response = client.get('api/v1/resources?page_size=10')
+    assert (len(response.json['data']) == 10)
+    response = client.get('api/v1/resources?page_size=100')
+    assert (len(response.json['data']) == 100)
+
+    # Test pages different and sequential
+    first_page_resource = response.json['data'][0]
+    assert (first_page_resource.get('id') == 1)
+    response = client.get('api/v1/resources?page_size=100&page=2')
+    second_page_resource = response.json['data'][0]
+    assert (second_page_resource.get('id') == 101)
+    response = client.get('api/v1/resources?page_size=100&page=3')
+    third_page_resource = response.json['data'][0]
+    assert (third_page_resource.get('id') == 201)
+
+    # Test bigger than max page size
+    too_long = PaginatorConfig.max_page_size + 1
+    response = client.get(f"api/v1/resources?page_size={too_long}")
+    assert (len(response.json['data']) == PaginatorConfig.max_page_size)
+
+    # Test farther than last page
+    too_far = 99999999
+    response = client.get(f"api/v1/resources?page_size=100&page={too_far}")
+    assert (len(response.json['data']) == 0)
```

Comment:
Should this be an error (404, perhaps?) instead of returning an empty list? We may have already talked about this, but I feel like it's worth reconsidering, especially for consistency across the API. If you request an ID that's too high, you get a 404. If you request a page that's too high, maybe you also should get a 404?